From 6df1559af7cd65e3faf7c61a2bb8f02667767ad6 Mon Sep 17 00:00:00 2001
From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>
Date: Wed, 31 Jul 2019 15:08:12 +0100
Subject: [PATCH 3/6] x86 machine types: Fixup dynamic sysbus entries

RH-Author: Dr. David Alan Gilbert <dgilbert@redhat.com>
Message-id: <20190731150814.29571-4-dgilbert@redhat.com>
Patchwork-id: 89814
O-Subject: [RHEL-AV-8.1.0 qemu-kvm PATCH v3 3/5] x86 machine types: Fixup dynamic sysbus entries
Bugzilla: 1719649
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>

From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>

We're missing a couple of upstream changes, add them for
consistency:

   v2.11.0-824-gef18310d54   Shouldn't have any effect
   v2.12.0-1411-g94692dcd71  Should allow us to use RAMFB if we enable
                             it

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/i386/pc_piix.c | 1 +
 hw/i386/pc_q35.c  | 4 +++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index c86c48c..3b9ba95 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -1037,6 +1037,7 @@ static void pc_machine_rhel7_options(MachineClass *m)
     pcmc->default_nic_model = "e1000";
     m->default_display = "std";
     m->no_parallel = 1;
+    machine_class_allow_dynamic_sysbus_dev(m, TYPE_RAMFB_DEVICE);
     compat_props_add(m->compat_props, pc_rhel_compat, pc_rhel_compat_len);
     m->alias = "pc";
     m->is_default = 1;
diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index 068813d..edf8e54 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -554,7 +554,9 @@ static void pc_q35_machine_rhel_options(MachineClass *m)
     m->default_display = "std";
     m->no_floppy = 1;
     m->no_parallel = 1;
-    machine_class_allow_dynamic_sysbus_dev(m, TYPE_SYS_BUS_DEVICE);
+    machine_class_allow_dynamic_sysbus_dev(m, TYPE_AMD_IOMMU_DEVICE);
+    machine_class_allow_dynamic_sysbus_dev(m, TYPE_INTEL_IOMMU_DEVICE);
+    machine_class_allow_dynamic_sysbus_dev(m, TYPE_RAMFB_DEVICE);
     m->alias = "q35";
     m->max_cpus = 384;
     compat_props_add(m->compat_props, pc_rhel_compat, pc_rhel_compat_len);
-- 
1.8.3.1

